### PR TITLE
Feature/rdsssam 155 messages output from the native samvera format should be translated

### PR DESCRIPTION
**Pull request by blackrat**
Big commit, this one.

Turned off geoblacklight, added a new gem and cherry picked some of the organisation stuff to make the messaging work. Lots of files, lots of comments describing them but essentially it is one file per message endpoint and a value translation to the model attribute.

### CODE REVIEW

File: willow/Gemfile [hunk 1]
```diff
@@ -26,6 +26,8 @@ gem 'jbuilder', '~> 2.5'
 gem 'hyrax', github: 'cottagelabs/hyrax'
 
 gem 'blacklight_advanced_search', '~> 6.0'
+gem 'blackrat_yaml_config'
```
Comment (dannyberzon) [inline willow/Gemfile]: I'm sure it's fine, but I'm a little nervous about adding this dependency to your repo if the repo has no license. Would it be a real hassle to add a gpl?

Comment (blackrat) [reply willow/Gemfile]: Anything I have is licensed in my contract anyway. I'm happy with adding a licence to it too though.

Comment (dannyberzon) [reply willow/Gemfile]: No need, I'm sure it's fine if that's the case :)

Comment (blackrat) [reply willow/Gemfile]: It's also covered by the Github T&Cs and I'll probably go for the MIT licence in any case. I wanted to make it just yaml_config, but that's already been parked about 7 years ago and I thought I could make this one a bolt-in replacement for that one eventually.

File: willow/app/models/cdm/messaging/enumerations.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        personIdentifierType
+        personRole
+        preservationEventType
+        relationType
+        resourceType
+        storageStatus
+        storageType
+        uploadStatus
+      ).each do |enum_section|
+        ::Cdm::Messaging::Enumerations::const_set(enum_section.classify, Decoder.(enum_section))
```
Comment (dannyberzon) [inline willow/app/models/cdm/messaging/enumerations.rb]: Yeah this is a good compromise on what we talked about. If I understand this correctly, once this module has been loaded, there will be a bunch of classes available `::Cdm::Messaging::Enumerations::AccessType` `::Cdm::Messaging::Enumerations::ChecksumType`

File: willow/app/models/cdm/messaging/enumerations/decoder.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+#
+#           def value3
+#             3
+#           end
+#         end
+#       end
+#     end
+#   end
+# end
+#
```
Comment (dannyberzon) [inline willow/app/models/cdm/messaging/enumerations/decoder.rb]: Just for neatness, might be nice to remove the commented out stuff above

File: willow/config/authorities/licenses.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 terms:
-    - id: http://opendefinition.org/licenses/cc-zero/
+    - id: http/opendefinition.org/licenses/cc-zero/
```
Comment (dannyberzon) [inline willow/config/authorities/licenses.yml]: What is this change? The new url is broken

File: willow/config/authorities/licenses.yml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
       term: Creative Commons CC Zero License (cc-zero)
       active: true
     - id: https://creativecommons.org/publicdomain/zero/1.0/
       term: Open Data Commons Public Domain Dedication and Licence (ODC PDDL)
       active: true
     - id: https://creativecommons.org/licenses/by/4.0/
-      term: Creative Commons Attribution License (cc-by)
+      term: Creative Commons Attribution License (CC BY 4.0)
```
Comment (dannyberzon) [inline willow/config/authorities/licenses.yml]: Why are we changing this? The Available license enumeration was agreed with Dom in this ticket:

https://jiscdev.atlassian.net/browse/RDSS-1651

The list Dom provided gave "(cc-by)" as the description for this license

File: willow/app/models/cdm/messaging/enumerations.rb (hunk 2 above)
Comment (blackrat) [reply willow/app/models/cdm/messaging/enumerations.rb]: Yes. Spot on.

File: willow/config/authorities/licenses.yml (hunk 5 above)
Comment (blackrat) [reply willow/config/authorities/licenses.yml]: It was in reviewing it with Dom today that he noted that it was wrong and should be caps and contain 4.0.

File: willow/app/models/cdm/messaging/enumerations/decoder.rb (hunk 3 above)
Comment (blackrat) [reply willow/app/models/cdm/messaging/enumerations/decoder.rb]: That's not commented out code. It's an example as to what the virtualized class would actually look like.

File: willow/config/authorities/licenses.yml (hunk 5 above)
Comment (dannyberzon) [reply willow/config/authorities/licenses.yml]: Perfect, thanks for clarifying

File: willow/app/models/cdm/messaging/enumerations/decoder.rb (hunk 3 above)
Comment (dannyberzon) [reply willow/app/models/cdm/messaging/enumerations/decoder.rb]: apologies

File: willow/config/authorities/licenses.yml (hunk 4 above)
Comment (blackrat) [reply willow/config/authorities/licenses.yml]: Not sure what happened there. Fixed.

File: willow/config/authorities/licenses.yml (hunk 5 above)
Comment (blackrat) [reply willow/config/authorities/licenses.yml]: Some others may have to change, so it's probably a good idea to have a ticket to update the text of them based on the link the ids take us to.

File: willow/app/models/cdm/messaging/enumerations/decoder.rb (hunk 3 above)
Comment (blackrat) [reply willow/app/models/cdm/messaging/enumerations/decoder.rb]: No worries.

**Pull request merged**